src/tbx/mod.rs

```diff
@@ -0,0 +1,410 @@
+// Copyright 2018 Manuel Holtgrewe, Berlin Institute of Health.
+// Licensed under the MIT license (http://opensource.org/licenses/MIT)
+// This file may not be copied, modified, or distributed
+// except according to those terms.
+
+use std::ffi;
+use std::mem;
+use std::ptr;
+use std::path::Path;
+use url::Url;
+use libc;
+
+use htslib;
+
+/// A trait for a Tabix reader with a read method.
+pub trait Read: Sized {
+    /// Read next line the given `String`.
+    /// Use this method in combination with a single allocated record to avoid the reallocations
+    /// occurring with the iterator.
+    ///
+    /// # Arguments
+    ///
+    /// * `record` - the `String` to be filled
+    fn read(&mut self, record: &mut String) -> Result<(), ReadError>;
+
+    /// Iterator over the lines/records of the seeked region.
+    /// Note that, while being convenient, this is less efficient than pre-allocating a
+    /// `String` and reading into it with the `read` method, since every iteration involves
+    /// the allocation of a new `String`.
+    fn records(&mut self) -> Records<Self>;
+
+    /// Return the text headers, split by line.
+    fn header(&self) -> &Vec<String>;
+}
+
+/// A Tabix file reader.
+#[derive(Debug)]
+pub struct TabixReader {
+    /// The header lines (if any).
+    header: Vec<String>,
+    /// The sequence names.
+    seqnames: Vec<String>,
```

Comment:
Are the seqnames really needed?

Reply:
Actually, this can be moved to a function. I don't know whether it's possible to get the types and life times correct to prevent copying and still have a correct program...